sd-cmd.go

```diff
@@ -28,46 +30,54 @@ func init() {
 	config.LoadConfig()
 }
 
+func runExecutor(sdAPI api.API, args []string) error {
+	err := executor.StartLog(args)
+	defer executor.FinishLog()
+
+	if err != nil {
+		return err
+	}
+	exec, err := executor.New(sdAPI, args)
+	if err != nil {
+		return err
+	}
+	err = exec.Run()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
 func runCommand(sdAPI api.API, args []string) error {
 	switch args[1] {
 	case "exec":
-		executor, err := executor.New(sdAPI, args[2:])
-		if err != nil {
-			return fmt.Errorf("Failed to create executor: %v", err)
-		}
-		output, err := executor.Run()
-		if err != nil {
-			fmt.Println(string(output))
-			return fmt.Errorf("Failed to run exec command: %v", err)
-		}
-		fmt.Println(string(output))
-		return nil
+		return runExecutor(sdAPI, args)
 	case "publish":
 		return fmt.Errorf("publish is not implemented yet")
 	case "promote":
 		return fmt.Errorf("promote is not implemented yet")
 	default:
-		return fmt.Errorf("No such type of command")
+		return runExecutor(sdAPI, args)
 	}
 }
 
 func main() {
 	defer finalRecover()
 
-	if len(os.Args) < 3 {
-		fmt.Printf("The argument num is not enough\n")
+	if len(os.Args) < 2 {
```

Comment:
Could you move this check into the function ```runCommand```?

I think it enables to reduce the surplus log.Prinln and os.Exit.

Reply:
Ok. I will move this one to runCommand function.